File: japiwznm/WznmQVerList.java
/**
  * \file WznmQVerList.h
  * Java API code for record of table TblWznmQVerList
  * \author Alexander Wirthmueller
  * \date created: 25 Aug 2020
  * \date modified: 25 Aug 2020
  */

package apiwznm;

import org.w3c.dom.*;
import sbecore.*;

public class WznmQVerList {
	
	public WznmQVerList(
				int jnum
				, String stubGrp
				, String stubOwn
				, String stubPrjRefWznmMProject
				, int Major
				, int Minor
				, int Sub
				, String stubBvrRefWznmMVersion
				, String stubRefWznmMLocale
				, String srefIxVState
				, String titIxVState
				, String srefsIxWDbmstype
				, String titsIxWDbmstype
			) {
		this.jnum = jnum;
		this.stubGrp = stubGrp;
		this.stubOwn = stubOwn;
		this.stubPrjRefWznmMProject = stubPrjRefWznmMProject;
		this.Major = Major;
		this.Minor = Minor;
		this.Sub = Sub;
		this.stubBvrRefWznmMVersion = stubBvrRefWznmMVersion;
		this.stubRefWznmMLocale = stubRefWznmMLocale;
		this.srefIxVState = srefIxVState;
		this.titIxVState = titIxVState;
		this.srefsIxWDbmstype = srefsIxWDbmstype;
		this.titsIxWDbmstype = titsIxWDbmstype;
	};

	public int jnum;
	public String stubGrp;
	public String stubOwn;
	public String stubPrjRefWznmMProject;
	public int Major;
	public int Minor;
	public int Sub;
	public String stubBvrRefWznmMVersion;
	public String stubRefWznmMLocale;
	public String srefIxVState;
	public String titIxVState;
	public String srefsIxWDbmstype;
	public String titsIxWDbmstype;
	
	public boolean readXML(
				Document doc
				, String basexpath
				, boolean addbasetag
			) {
		if (addbasetag) basexpath = Xmlio.checkUclcXPaths(doc, basexpath, "WznmQVerList");

		if (Xmlio.checkXPath(doc, basexpath)) {
			stubGrp = Xmlio.extractStringUclc(doc, basexpath, "stubGrp", "grp", null, 0);
			stubOwn = Xmlio.extractStringUclc(doc, basexpath, "stubOwn", "own", null, 0);
			stubPrjRefWznmMProject = Xmlio.extractStringUclc(doc, basexpath, "stubPrjRefWznmMProject", "prj", null, 0);
			Major = Xmlio.extractIntegerUclc(doc, basexpath, "Major", "maj", null, 0);
			Minor = Xmlio.extractIntegerUclc(doc, basexpath, "Minor", "min", null, 0);
			Sub = Xmlio.extractIntegerUclc(doc, basexpath, "Sub", "sub", null, 0);
			stubBvrRefWznmMVersion = Xmlio.extractStringUclc(doc, basexpath, "stubBvrRefWznmMVersion", "bvr", null, 0);
			stubRefWznmMLocale = Xmlio.extractStringUclc(doc, basexpath, "stubRefWznmMLocale", "loc", null, 0);
			srefIxVState = Xmlio.extractStringUclc(doc, basexpath, "srefIxVState", "ste", null, 0);
			titIxVState = Xmlio.extractStringUclc(doc, basexpath, "titIxVState", "ste2", null, 0);
			srefsIxWDbmstype = Xmlio.extractStringUclc(doc, basexpath, "srefsIxWDbmstype", "dty", null, 0);
			titsIxWDbmstype = Xmlio.extractStringUclc(doc, basexpath, "titsIxWDbmstype", "dty2", null, 0);
			
			return true;
		};

		return false;
	};
};

File: apiwznm/WznmQVerList.cpp
/**
	* \file WznmQVerList.cpp
	* API code for table TblWznmQVerList (implementation)
	* \copyright (C) 2016-2020 MPSI Technologies GmbH
	* \author Alexander Wirthmueller (auto-generation)
	* \date created: 5 Dec 2020
  */
// IP header --- ABOVE

#include "WznmQVerList.h"

using namespace std;
using namespace Sbecore;
using namespace Xmlio;

/******************************************************************************
 class WznmQVerList
 ******************************************************************************/

WznmQVerList::WznmQVerList(
			const uint jnum
			, const string stubGrp
			, const string stubOwn
			, const string stubPrjRefWznmMProject
			, const usmallint Major
			, const usmallint Minor
			, const usmallint Sub
			, const string stubBvrRefWznmMVersion
			, const string stubRefWznmMLocale
			, const string srefIxVState
			, const string titIxVState
			, const string srefsIxWDbmstype
			, const string titsIxWDbmstype
		) :
			jnum(jnum)
			, stubGrp(stubGrp)
			, stubOwn(stubOwn)
			, stubPrjRefWznmMProject(stubPrjRefWznmMProject)
			, Major(Major)
			, Minor(Minor)
			, Sub(Sub)
			, stubBvrRefWznmMVersion(stubBvrRefWznmMVersion)
			, stubRefWznmMLocale(stubRefWznmMLocale)
			, srefIxVState(srefIxVState)
			, titIxVState(titIxVState)
			, srefsIxWDbmstype(srefsIxWDbmstype)
			, titsIxWDbmstype(titsIxWDbmstype)
		{
};

bool WznmQVerList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "WznmQVerList");
	else
		basefound = checkXPath(docctx, basexpath);

	if (basefound) {
		extractStringUclc(docctx, basexpath, "stubGrp", "grp", stubGrp);
		extractStringUclc(docctx, basexpath, "stubOwn", "own", stubOwn);
		extractStringUclc(docctx, basexpath, "stubPrjRefWznmMProject", "prj", stubPrjRefWznmMProject);
		extractUsmallintUclc(docctx, basexpath, "Major", "maj", Major);
		extractUsmallintUclc(docctx, basexpath, "Minor", "min", Minor);
		extractUsmallintUclc(docctx, basexpath, "Sub", "sub", Sub);
		extractStringUclc(docctx, basexpath, "stubBvrRefWznmMVersion", "bvr", stubBvrRefWznmMVersion);
		extractStringUclc(docctx, basexpath, "stubRefWznmMLocale", "loc", stubRefWznmMLocale);
		extractStringUclc(docctx, basexpath, "srefIxVState", "ste", srefIxVState);
		extractStringUclc(docctx, basexpath, "titIxVState", "ste2", titIxVState);
		extractStringUclc(docctx, basexpath, "srefsIxWDbmstype", "dty", srefsIxWDbmstype);
		extractStringUclc(docctx, basexpath, "titsIxWDbmstype", "dty2", titsIxWDbmstype);
	};

	return basefound;
};

/******************************************************************************
 class ListWznmQVerList
 ******************************************************************************/

ListWznmQVerList::ListWznmQVerList() {
};

ListWznmQVerList::ListWznmQVerList(
			const ListWznmQVerList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WznmQVerList(*(src.nodes[i]));
};

ListWznmQVerList& ListWznmQVerList::operator=(
			const ListWznmQVerList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WznmQVerList(*(src.nodes[i]));

	return *this;
};

ListWznmQVerList::~ListWznmQVerList() {
	clear();
};

void ListWznmQVerList::clear() {
	for (unsigned int i = 0; i < nodes.size(); i++) if (nodes[i]) delete nodes[i];
	nodes.resize(0);
};

bool ListWznmQVerList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	vector<unsigned int> ics;
	vector<bool> shorttags;

	WznmQVerList* rec = NULL;

	string s;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "ListWznmQVerList");
	else
		basefound = checkXPath(docctx, basexpath);

	extractList(docctx, basexpath, "WznmQVerList", "row", "jnum", ics, shorttags);

	clear();

	for (unsigned int i = 0; i < ics.size(); i++) {
		rec = new WznmQVerList(ics[i]);

		if (shorttags[i]) s = basexpath + "/row[@jnum='" + to_string(ics[i]) + "']";
		else s = basexpath + "/WznmQVerList[@jnum='" + to_string(ics[i]) + "']";

		if (rec->readXML(docctx, s)) nodes.push_back(rec);
		else delete rec;
	};

	return basefound;
};
